src/backend/opencl/platform.hpp

```diff
@@ -198,5 +201,8 @@ class DeviceManager
         std::unique_ptr<GraphicsResourceManager> gfxManagers[MAX_DEVICES];
 #endif
         std::unique_ptr<clfftSetupData> mFFTSetup;
+
+        using BoostProgCache = boost::shared_ptr<boost::compute::program_cache>;
+        std::vector<BoostProgCache*> mBoostProgCacheVector;
```

Comment:
Since we are using a pointer here. Can you use a forward declaration instead of including the headers. I want to minimize the includes in this header since it is used everywhere.

Reply:
Addressed